src/app/login/page.tsx

```diff
@@ -0,0 +1,164 @@
+			loginUser(id, () => {
+				router.push("/");
+			});
+		});
+	};
+
+	/* Render ----------------------------------------------------------------------------------------------------------------------------------------------------------------*/
+
+	return (
+		<section className="w-full h-[calc(100vh-76px)] flex justify-center items-center">
+			<Image
+				src={"/login.png"}
+				alt="Login Image"
+				className="lg:inline hidden flex-1 w-1/2 h-[calc(100vh-76px)] overflow-hidden bg-black object-cover"
+				height={100}
+				width={100}
+			/>
+
+			<div className="lg:mx-4 flex flex-1 flex-col justify-center items-center">
+				<div className="flex flex-1 flex-col justify-center items-center max-w-xs w-full">
+					<h2 className="font-semibold text-2xl">
+						{showLogin ? "Log in" : "Create an account"}
+					</h2>
+
+					{!showLogin && (
+						<p className="text-gray-400 text-sm mt-3">
+							Enter your email below to create your account
+						</p>
+					)}
+
+					<div className="mt-4 w-full">
+						<input
+							type="email"
+							placeholder="name@example.com"
+							className="w-full border-[1px] px-4 py-1 rounded-md shadow-sm"
+							value={email}
+							onChange={(e) => setEmail(e.target.value)}
+						/>
+					</div>
+					<div className="mt-4 w-full">
+						<input
+							type="password"
+							placeholder="password"
+							className="w-full border-[1px] px-4 py-1 rounded-md shadow-sm"
+							value={password}
+							onChange={(e) => setPassword(e.target.value)}
+						/>
+					</div>
+					{!showLogin && (
+						<div className="mt-4 w-full">
+							<input
+								type="password"
+								placeholder="confirm password"
+								className="w-full border-[1px] px-4 py-1 rounded-md shadow-sm"
+								value={confirmPassword}
+								onChange={(e) =>
+									setConfirmPassword(e.target.value)
+								}
+							/>
+						</div>
+					)}
+					{!showLogin && (
+						<div className="mt-4 w-full">
+							<input
+								placeholder="username"
+								className="w-full border-[1px] px-4 py-1 rounded-md shadow-sm"
+								value={username}
+								onChange={(e) => setUsername(e.target.value)}
+							/>
+						</div>
+					)}
+					<div className="w-full mt-4">
+						<Button
+							label="Continue"
+							callback={() =>
+								showLogin ? handleLogin() : handleSignUp()
+							}
+							full
+						/>
+					</div>
+
+					<p className="mt-4 text-sm text-gray-400 text-center px-8">
+						{showLogin
+							? "Don't have an account?"
+							: "Already a user?"}
+						&nbsp;
+						<span
+							className="underline cursor-pointer"
+							onClick={() => setShowLogin(!showLogin)}
+						>
+							{showLogin ? "Sign up" : "Sign in"}
+						</span>
+					</p>
+					<p className="mt-4 text-sm text-gray-400 text-center px-8">
+						By clicking continue, you agree to our
+						<span> Terms of Service </span>
+						and
+						<span> Privacy Policy</span>.
+					</p>
+				</div>
+			</div>
+		</section>
+	);
+};
```

Comment:
Could we possibly create 2 separate pages here? 1 for login and 1 for signup instead of combining both? To me it'd be simpler and easier to read.